Persistence features for Runtime

Incoming5643 opened the pull request:
- Runtime now keeps track of all cats in the world, at the start of the next round any cats that survived the previous one will spawn with Runtime in the CMO's office.
- After 9 deaths Runtime will be mysteriously missing from the next station (but only for that round). This also clears out any family Runtime might have.

---

A note on mechanics: Yes this allows for potentially infinite cats, but in reality this is not a practical problem because:
1. Runtime and Proc can have kittens, and kittens are saved BUT they can only have three. This is true even if they start with three kittens they didn't actually have. It's technically possible to get more kittens by repeatedly removing new kittens from the cat breeding chamber, but that's a lot of work just to catsplosion the CMO next round.
2. The petsplosion wizard event can exacerbate a cat situation, but Runtime won't carry over other versions of herself, and since Runtime is the only female cat in the whole dang game this puts a bottleneck on kittens that should keep things from going nuts. 
3. It only takes one successful nuke round to genocide the cats.
4. :cat2:

Incoming5643 committed:
Persistence features for Runtiem

* Runtime now keeps track of all cats in the world, at the start of the next round any cats that survived the previous one will spawn with runtime in the CMO's office. This mechanic hinges on Runtime, so if she dies no other cats will spawn next round.

* After 9 death Runtime will be mysteriously missing from the next station (but only for that round).

RemieRichards commented on code/modules/mob/living/simple_animal/friendly/cat.dm:
```diff
@@ -60,6 +60,60 @@
+
+	for(var/cat_type in family)
+		if(family[cat_type] > 0)
+			for(var/i = 0, i < family[cat_type], i++)
```
```for(var/i in 0 to family[cat_type]-1)```
or
```for(var/i in 1 to family[cat_type])```

PKPenguin321 commented:
You should add a line to it's examine text saying how many lives it has.
something like,
"If cats truly have nine lives, then this looks like it must have died [once/twice/three times/four times etc.] already."

Zilenan91 commented:
What about kittens in the pet boxes in mining artifact rooms, will those go to the CMO's office

Incoming5643 commented:
Animals in cargo crates don't actually spawn until they're opened, so they won't be saved.

The holodeck cats are definitely an open question though

Zilenan91 commented:
It's probably fine because I can call the CMO a crazy cat lady

KorPhaeron commented:
lmao2cat

Niknakflak commented:
You are driving this into the ground tbh
It made sense with poly but now its just meh.
It's cat PR so :+1: but eh

Incoming5643 commented:
I want to give every pet a thing so no one feels left out.

Except maybe that fox, fuck him.

Razharas commented:
WHen will we be able to give runtime an antag role?
@Incoming5643

bgobandit commented:
I support cat functionality

Aloraydrel commented:
A FELINE PAIR!!

bgobandit commented:
I just got the pun in the branch name

you devil you

duncathan commented:
:+1:

Iamgoofball commented:
@Incoming5643 

Cats should follow around the Janicart if the keys are in and it's being driven around.

Shadowlight213 commented:
@incoming Lamarr expansion when?

Incoming5643 committed:
Cleanup pull

Requests heeded.
Holodeck loophole somewhat hamfistedly fixed.
Bonus: One stealth homestuck reference removed.

Incoming5643 committed:
Damn you web editor

Incoming5643 commented:
:cat2: Should be good to go now :cat2:

duncathan commented on code/modules/mob/living/simple_animal/friendly/cat.dm:
```diff
@@ -54,12 +54,68 @@
 /mob/living/simple_animal/pet/cat/Runtime
 	name = "Runtime"
-	desc = "GCAT"
+	desc = "Tends to show up in the strangest of places."
```
noooo put it back put it back ;_;

Incoming5643 committed:
Fine just for you

Incoming5643 commented:
just for you @duncathan

RemieRichards commented:
I'm disappointed you tried to remove the GCAT description.
its literally the smartest joke we have.

Incoming5643 commented:
Why did you have to say that now I gotta code dive descriptions to make sure your right.
